src/sentry/integrations/pagerduty/integration.py

```diff
@@ -0,0 +1,185 @@
+    source_url="https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/pagerduty",
+    aspects={},
+)
+
+
+class PagerDutyIntegration(IntegrationInstallation):
+    def get_client(self):
+        pass
+
+    def get_organization_config(self):
+        from sentry.models import Project
+
+        projects = Project.objects.filter(
+            organization_id=self.org_integration.organization_id,
+        )
+        items = []
+        for p in projects:
+            items.append({"value": p.id, "label": p.name})
+
+        service_options = [(s.id, s.service_name,) for s in self.services]
+
+        fields = [
+            {
+                "name": "project_mapping",
+                "type": "choice_mapper",
+                "label": "Map projects in Sentry to services in PagerDuty",
+                "help": "When an alert rule is triggered in a project, this mapping will let us know what service to create the incident in PagerDuty.",
+                "addButtonText": "Add Sentry Project",
+                "addDropdown": {
+                    "emptyMessage": "All projects configured",
+                    "noResultsMessage": "Could not find project",
+                    "items": items,
+                },
+                "mappedSelectors": {
+                    "service": {
+                        "choices": service_options,
+                        "placeholder": "Select a service",
+                    }
+                },
+                "columnLabels": {"service": "Service"},
+                "mappedColumnLabel": "Sentry Project",
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data):
+
+        if "project_mapping" in data:
+            project_ids_and_services = data.pop("project_mapping")
+
+            with transaction.atomic():
+                PagerDutyServiceProject.objects.filter(
+                    pagerduty_service__in=PagerDutyService.objects.filter(organization_integration=self.org_integration),
+                ).delete()
+
+                for p_id, s in project_ids_and_services.items():
+                    # create the record in the table
+                    project = Project.objects.get(pk=p_id)
+                    service = PagerDutyService.objects.get(id=s["service"])
+                    PagerDutyServiceProject.objects.create(
+                        project=project,
+                        pagerduty_service=service,
+                    )
+
+    def get_config_data(self):
+        config = self.org_integration.config
+        project_mappings = PagerDutyServiceProject.objects.filter(
+            pagerduty_service__in=PagerDutyService.objects.filter(
+                organization_integration_id=self.org_integration.id,
+            ),
+        )
+        data = {}
+        for pm in project_mappings:
+            data[pm.project_id] = {
+                "service": pm.pagerduty_service.id,
```

Comment:
Could the `pagerduty_service` be eagerloaded so we don't have an N+1 query?

Reply:
actually all I need is `pm.pagerduty_service_id`